### PR TITLE
Created a function to get submission photo by its submission id

### PR DESCRIPTION
**Pull request by Sujanadh**
## Updates:
This PR updates the `OdkCentral` creating a new function `getSubmissionPhoto` to get the photo content from the individual submission.
- it taskes parameters such as odk project id, submission id(instance id) , xform(form id), filename (name of photo file)
- it fetches the media content in bytes which can be used directly to download or upload anywhere.

### CODE REVIEW

File: osm_fieldwork/OdkCentral.py [hunk 1]
```diff
@@ -806,6 +806,33 @@ def getSubmissionMedia(
...
+
+        Returns:
+            (bytes): The media data
+        """
+        url = f"{self.base}projects/{projectId}/forms/{xform}/submissions/{instanceID}/attachments/{filename}"
+        result = self.session.get(url, verify=self.verify)
+        if result.status_code == 200:
+            log.debug(f"fetched {filename} from Central")
+        else:
+            status = eval(result._content)
```
Comment (spwoodcock) [inline osm_fieldwork/OdkCentral.py:832]: It's generally a bad idea to use `eval` in Python - there is almost always a better way!

Also accessing a private method is generally bad too.

Can this be achieved using .json() or another method?

**Review by spwoodcock — CHANGES_REQUESTED** [review]
It would be really nice if this included a test too 😁

But not essential if it will take lots of time 👍

File: osm_fieldwork/OdkCentral.py (hunk 1 above)
Comment (Sujanadh) [reply osm_fieldwork/OdkCentral.py:832]: In `getMedia` func too eval was used, so I thought to use the same, but you are right, `eval` is not always a safe option. will refactor for both functions.

**Comment by Sujanadh** [comment]
> It would be really nice if this included a test too 😁
> 
> But not essential if it will take lots of time 👍

how to use dummy submission photo?

**Comment by spwoodcock** [comment]
> > It would be really nice if this included a test too 😁
> > 
> > But not essential if it will take lots of time 👍
> 
> how to use dummy submission photo?

You could possibly just upload some known bytes, not an actual photo:

`b'fakephotodata'`

I assume uploading this with a `jpg` extension would work and would be lightweight (unless Central has a check to see if it's an actual image!)

Then when you check the download, `assert photo_data == b'fakephotodata'`

**Comment by spwoodcock** [comment]
Tests aren't essential here, just a nice to have!

We can merge this - bump to remind myself!

**Comment by Sujanadh** [comment]
thanks for reminding!

**Comment by spwoodcock** [comment]
I'll merge this for the next release 👍 

If we have tests in future, they don't need to be in a release, just present on `main`, so they can come later

File: osm_fieldwork/OdkCentral.py [hunk 2]
```diff
@@ -829,7 +829,7 @@ def getSubmissionPhoto(
         if result.status_code == 200:
             log.debug(f"fetched {filename} from Central")
         else:
-            status = eval(result._content)
+            status = json.loads(result.content)
```
Comment (spwoodcock) [inline osm_fieldwork/OdkCentral.py:832]: Actually, sorry to be a pain, but using `result.json()` from the requests module is more efficient here! It also saves the import of the json module 😄

**Review by spwoodcock — CHANGES_REQUESTED** [review]

**Pull request merged**